examples/dyn_proxy.rs

```diff
@@ -0,0 +1,59 @@
+#![deny(warnings)]
+use warp::hyper::body::{Body, Bytes};
+use warp::hyper::{Client, Request};
+use warp::{
+    http::{method::Method, HeaderMap, Response},
+    path::FullPath,
+    Filter, Rejection,
+};
+
+static PROXY_TARGET: &'static str = "http://httpbin.org";
+
+async fn proxy_request(
+    method: Method,
+    path: FullPath,
+    headers: HeaderMap,
+    body: Bytes,
+) -> Result<Response<Body>, Rejection> {
+    let request = build_request(&method, &path, &headers, body);
+    let client = Client::new();
+    let response = client.request(request).await.unwrap();
+    let response_status = response.status();
+    let response_body = response.into_body();
+
+    Ok(Response::builder()
+        .status(response_status)
+        .body(response_body)
+        .unwrap())
+}
+
+fn build_request(
+    method: &Method,
+    path: &FullPath,
+    headers: &HeaderMap,
+    body: Bytes,
+) -> Request<Body> {
+    let uri = format!("{}/{}", PROXY_TARGET, path.as_str());
+
+    let mut request = Request::builder().method(method.as_str()).uri(uri);
+
+    for (key, value) in headers {
+        request = request.header(key, value);
+    }
+
+    request.body(Body::from(body)).unwrap()
+}
+
+#[tokio::main]
+async fn main() {
+    let routes = warp::method()
+        .and(warp::path::full())
+        .and(warp::header::headers_cloned())
+        .and(warp::body::bytes())
```

Comment:
You could use `warp::body::steam()`, so you don't need to buffer the body in the proxy, but just stream it along.

Reply:
I tried to use `warp::body::stream()` here, but i could not figure out how to send a `client.request` with a `Stream`. It seems that `client.requests` expects a type that implements `HttpBody + Send + 'static` but i could not convert the stream to work it out. I might be missing something here?

Reply:
Hi @cars10,
something like:
```Rust
fn build_request(
    method: Method,
    path: FullPath,
    headers: HeaderMap,
    body: impl Stream<Item = Result<impl Buf, Error>> + Send + 'static,
) -> Request<Body> {
    let uri = format!("{}/{}", PROXY_TARGET, path.as_str());

    let body = body.map_ok(|mut buf| buf.copy_to_bytes(buf.remaining()));
    let mut request = Request::new(Body::wrap_stream(body));
    *request.method_mut() = method;
    *request.uri_mut() = uri.parse().unwrap();
    *request.headers_mut() = headers;
    request
}
```

Reply:
Done! Thanks for the hint.